File: mapHandler.js
//x, y format
var node = [
 178, 936,
 325, 806,
 325, 616,
 383, 433,
 352, 345,
 163, 421,
 70, 432,
 266, 533,
 100, 298,
 260, 217,
 577, 186,
 683, 216,
 830, 204,
 1115, 160,
 1529, 116,
 1534, 265,
 1522, 442,
 1502, 579,
 1412, 723,
 1361, 927,
 948, 939,
 975, 792,
 488, 804,
 497, 951,
 947, 510,
 1173, 570,
 1076, 803,
 1318, 771,
 159, 805,
 733, 381
];

//index of pairs of nodes, represent  edges
var edge =[
1, 24,
1, 29,
2, 3,
2, 6,
2, 7,
2, 8,
2, 23,
2, 29,
3, 7,
3, 8,
3, 29,
4, 5,
4, 8,
5, 6,
5, 12,
6, 7,
6, 8,
6, 29,
7,  8,
7, 9,
7, 29,
8, 29,
9, 10,
10, 11,
11, 12,
12, 13,
12, 30, 
13, 14,
13, 30,
14, 15,
14, 16,
15, 16,
16, 17,
17, 18,
18, 19,
18, 26,
19, 20,
19, 26,
19, 27,
19, 28,
20, 21,
21, 22,
21, 24,
22, 23,
22, 25,
22, 27,
23, 24,
25, 26,
26, 28,
27, 28
];

var MapHandler = {
    dict : new Map(),
    dis_mtx : [],
    path_mtx : [],
    init : function(node, edge){
        var i, j, temp;
        for (i=0; i < node.length; i+=2){
            temp = [node[i], node[i+1]];
            this.dict.set(i >> 1, [temp, []]);
        }
        this.nodes_num = node.length >> 1;
        for (i=0; i < this.nodes_num; i++){
            this.dis_mtx[i] = [];
            this.path_mtx[i] = [];
        }
        for (i=0; i < this.nodes_num; i++){
            for (j=0; j < this.nodes_num; j++){
                this.dis_mtx[i][j] = Infinity;
                this.path_mtx[i][j] = null;
            }
        }
       var node1, node2, distance;
       for (j=0; j < edge.length; j+=2){
           node1 = edge[j] - 1;
           node2 = edge[j + 1] - 1;
           distance = this.get_distance(node1, node2);
           this.dict.get(node1)[1].push([node2, node1]);
           this.dict.get(node2)[1].push([node1, node2]);
           this.dis_mtx[node1][node2] = distance;
           this.dis_mtx[node2][node1] = distance;
       }
    },
    
    get_distance : function(node1, node2){
        var n1, n2;
        n1 = this.dict.get(node1)[0];
        n2 = this.dict.get(node2)[0];
        return ((n1[0] - n2[0])**2 + (n1[1] - n2[1])**2)**0.5;
    },
    
    get_min_path : function(){
        var k, i, j;
        for (k = 0; k < this.nodes_num; k++){
            for (i = 0; i < this.nodes_num; i++){
                for (j = 0; j < this.nodes_num; j++){
                    if (this.dis_mtx[i][j] > this.dis_mtx[i][k] + this.dis_mtx[k][j]){
                        this.dis_mtx[i][j] = this.dis_mtx[i][k] + this.dis_mtx[k][j];
                        this.path_mtx[i][j] = k;
                    }
                }
            }            
        }
    },
    
    get_one_path : function(node1, node2){
        node1 -= 1;
        node2 -= 1;
        var node = node1;
        var path = [];
        while(node != null){
            path.push(node + 1);
            node = this.path_mtx[node][node2];
        }
        path.push(node2 + 1);
        return path;
    }
}

MapHandler.init(node, edge);
MapHandler.get_min_path();
console.log(MapHandler.get_one_path(1, 14));

File: mapHandler.py
from newData import *
class Map():
    def __init__(self, node, edge):
        #data structure output in this case
        #{0:[(1, 4), [(1, 2.344), (3, 4.566), (5, 7.688), (6, 8.900)]], 1:[(2, 5),  ......}
        self.dict = {}
        for i in range(0, len(node), 2):
            temp = (node[i], node[i+1])
            self.dict[i >>1] = (temp,[])
        self.nodes_num = len(node) >> 1
        self.dis_mtx = [[float('inf') for j in range(self.nodes_num)] for i in range(self.nodes_num)]
        self.path_mtx = [[None for j in range(self.nodes_num)] for i in range(self.nodes_num)]
        
        for j in range(0, len(edge), 2):
            node1 = edge[j] - 1
            node2 = edge[j + 1] - 1
            assert node1 in self.dict
            assert node2 in self.dict
            distance = self.get_distance(node1, node2)
            self.dict[node1][1].append((node2, distance))
            self.dict[node2][1].append((node1, distance))
            self.dis_mtx[node1][node2] = distance
            self.dis_mtx[node2][node1] = distance
            self.path_mtx[node1][node2] = node2
            self.path_mtx[node2][node1] = node1
        
        self.current_location = None
        self.destination = None
       
        
        
    def set_destination(self, destination):
        assert destination in self.dict
        self.destination = destination
    
    def set_current_location(self, current_location):
        assert current_location in self.dict
        self.current_location = current_location
    
    def get_distance(self, node1, node2):
        n1 = self.dict[node1][0]
        n2 = self.dict[node2][0]
        return ((n1[0] - n2[0])**2 + (n1[1] - n2[1])**2)**0.5
        
    def get_min_path(self):
        for k in range(self.nodes_num):
            for i in range(self.nodes_num):
                for j in range(self.nodes_num):
                    if self.dis_mtx[i][j] > self.dis_mtx[i][k] + self.dis_mtx[k][j]:
                        self.dis_mtx[i][j] = self.dis_mtx[i][k] + self.dis_mtx[k][j]
                        self.path_mtx[i][j] = k
        
    def get_one_path(self, node1, node2):
        node1 -= 1
        node2 -= 1
        path = [node1 + 1]
        def get_path(self, node1, node2, path):
            if(node1 == node2):
                return
            if(self.path_mtx[node1][node2]==node2):
               path.append(node2 + 1)
            else:
                get_path(self, node1, self.path_mtx[node1][node2], path)
                get_path(self, self.path_mtx[node1][node2], node2, path)        
        get_path(self, node1, node2, path)
        return path
    
instance = Map(node, edge)
instance.get_min_path()
print(instance.get_one_path(1, 14))
